optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/AssortedConstraintStreamTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.score.stream;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+import org.optaplanner.core.impl.score.director.InnerScoreDirector;
+import org.optaplanner.core.impl.testdata.domain.score.lavish.TestdataLavishEntity;
+import org.optaplanner.core.impl.testdata.domain.score.lavish.TestdataLavishEntityGroup;
+import org.optaplanner.core.impl.testdata.domain.score.lavish.TestdataLavishSolution;
+
+import static org.optaplanner.core.api.score.stream.Joiners.equal;
+
+public class AssortedConstraintStreamTest extends AbstractConstraintStreamTest {
```

Comment:
Assorted?
I don't like "Assorted"Test. It's like Drools's notorious "Misc"Test (and Misc2Test and Misc3Test). It's like a "lost+find" directory - tests that haven't been sorted yet.

So far, the "where can I find it" logic in the tests has been:
if the final stream is a Uni stream, in the UniTest, if it's a Bi then in the BiTest, etc.
But if you think it makes more sense to put some of these grouping tests into a (Advanced)GroupingConstraintStreamTest, etc to avoid bloating the BiTest etc, I am ok with that.

Veto on naming a test "Assorted" or "Misc" Test. It's too broad.

Reply:
Yeah, this was one of the things I was going to "clean up later before I submit the PR." :-)